### PR TITLE
Bug fix: raise value error if gradient_accumulation_steps is zero, ne…

### PR DESCRIPTION
**Pull request by Arunprakash-A**
…gative or None

gradient_accumulation_steps has been used at multiple places, carrying out division and multiplication operations. It is safe to raise a Value Error to avoid unexpected behaviours if a negative integer passed.  For example, the gradient_accumulation_steps accepts the negative integer and produces a NEGATIVE valued loss

```
from transformers import TrainingArguments, Trainer
training_args = TrainingArguments("test-trainer",num_train_epochs=1, per_device_train_batch_size=1,
                                  disable_tqdm=None,gradient_accumulation_steps=-2) # passed negative int to grad_acc_steps
trainer = Trainer(
    model,
    training_args,
    train_dataset=tokenized_datasets["train"].select(range(10)),
    eval_dataset=tokenized_datasets["validation"].select(range(10)),
    data_collator=data_collator,
    tokenizer=tokenizer,
    callbacks=[TraceMemory],

)

result = trainer.train()
print(result)
```
Gives the following output

> TrainOutput(global_step=1, training_loss=-0.6691113710403442, metrics={'train_runtime': 0.3403, 'train_samples_per_second': 29.383, 'train_steps_per_second': 2.938, 'total_flos': 1864050241536.0, 'train_loss': -0.6691113710403442, 'epoch': 0.2})

Note that the **loss is negative**! This is an unexpected behaviour due to the negative integer assigned to gradient_accumulation_steps.

# What does this PR do?

Fixes a Bug

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -399,6 +399,9 @@ def __init__(
                     " boolean argument which will be triggered after the last batch of the eval set to signal that the"
                     " summary statistics should be returned by the function."
                 )
+        if not args.gradient_accumulation_steps or args.gradient_accumulation_steps < 1 :
+            raise ValueError(f"gradient_accumulation_steps in the TrainingArguments takes a positive integer, not {args.gradient_accumulation_steps}")
```
Comment (amyeroberts) [inline src/transformers/trainer.py:403]: This should be asserted in the training args init, rather than the trainer class

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding this! 

general comment on where this logic should go.

File: src/transformers/trainer.py (hunk 1 above)
Comment (Arunprakash-A) [reply src/transformers/trainer.py:403]: Yes, I completely agree. To me, the concerning factor is if it is set to a negative value (by mistake, or for whatever reason), the program runs and leads to **logical error**! If you agree, please add the checking in the TrainingArguments. I leave the decision to you!

Comment (amyeroberts) [reply src/transformers/trainer.py:403]: As setting this to a negative value is very unlikely I think we should just close  the PR to add the check
